src/main/kotlin/com/exactpro/th2/codec/grpc/GrpcCodecService.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2022-2022 Exactpro (Exactpro Systems Limited)
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.codec.grpc
+
+import com.exactpro.th2.common.grpc.AnyMessage
+import com.exactpro.th2.common.grpc.MessageGroupBatch
+import com.exactpro.th2.common.metrics.SESSION_ALIAS_LABEL
+import com.exactpro.th2.common.schema.grpc.router.GrpcRouter
+import io.grpc.stub.StreamObserver
+import mu.KotlinLogging
+
+class GrpcCodecService(
+    grpcRouter: GrpcRouter,
+    private val generalDecodeFunc: ((MessageGroupBatch) -> MessageGroupBatch),
+    private val generalEncodeFunc: ((MessageGroupBatch) -> MessageGroupBatch)
+) : CodecGrpc.CodecImplBase() {
+
+    private val nextCodec = try {
+        grpcRouter.getService(AsyncCodecService::class.java)
+    } catch (t: Throwable) {
+        null
+    }.apply {
+        LOGGER.info { "Next codec in pipeline stub: $this" }
```

Comment:
> $this

What does it print? I'm not sure that there will be something human-readable

Reply:
It prints stub class name: "com.exactpro.th2.codec.grpc.CodecDefaultAsyncImpl@21da4b5f"
or it prints "null". No more information is available here.